### DISCUSSION TITLE
"The Design is Very Human"

### DISCUSSION THREAD

**Discussion by snowboyz0825** [TurboWarp/extensions#1067] [Show and tell]
![block_9_30_2023-11_06_36 PM](https://github.com/TurboWarp/extensions/assets/101284660/d557a206-1eeb-40f5-87c5-f72ff409290a)

**Comment by snowboyz0825** [comment]
If my brain wasn't mush already, it is now. Why is the code for a block that immatitates the split block in lily's toolbox so much? It utilizes custom reporters, and it's 208 blocks. I encourage you to code it shorter, please, this can't possibly be the most effecient way to do this.

**Comment by softedco** [comment]
![image](https://github.com/TurboWarp/extensions/assets/100989385/a0e69140-981f-4bd5-a059-0fd100cb1d14)
I didn't use custom blocks to cut the amount of used blocks, I also used no variables and assumed the list is used somewhere else so it has no side effects for which I needed the return and free custom block, there's only 68 blocks

![image](https://github.com/TurboWarp/extensions/assets/100989385/f8f7eb93-53ee-4f30-9ecf-b0c1b91eba78)
here is the variant with variables, 38 blocks

**Reply by softedco** [reply]
it doesn't check for the item argument being less than zero but that's just 4 additional blocks

**Reply by snowboyz0825** [reply]
I should've been more specific with what I needed. I wanted it in one custom block, using a list used by several other custom blocks. Also, an error could happen if you put more than 1 character in the delimeter. I had an original version. 
![block_9_30_2023-8_20_45 PM](https://github.com/TurboWarp/extensions/assets/101284660/3d2ceb1a-a6c8-42d9-9ded-e5fb80bfbb9e)
that's basically the same thing. I'm using this to try and get rid of lily's toolbox in my project cause everyone hates it for some reason, and sometimed I have to split by more than one character, which is why it becomes so weird

**Reply by snowboyz0825** [reply]
Also how'd you get your blocks to look like that it's so much easier on the eyes

**Reply by CubesterYT** [reply]
> Also how'd you get your blocks to look like that it's so much easier on the eyes

@snowboyz0825 custom block colors in addon settings.

**Reply by snowboyz0825** [reply]
Yea I got that it's soooo much better

**Comment by softedco** [comment]
for the 1 custom block you can use a retval variable, I checked and it shouldn't cause any problems if multiple functions are using it, for the delimeter, just do (letter (1) of (delimeter)) or if you want to check for the whole delimeter I might show you later but it shouldn't be any more difficult. for the blocks, it's scratch addons / customizable block shape

**Reply by snowboyz0825** [reply]
My point was if you were to split say test1-split-test2 by -split-, it would get stuck searching for a -split- as a single character. My code was altered to check if letter one matches, then letter 2, then letter 3, etc, and it resets itself if the next letter doesn't match. Once you get all the letters matched, it changes what would essentially be the "item" by -1, and once it reaches 0 it starts to actually keep track of what it goes over. Once it gets to -1, due to it recording the next instance of a "split by" or gets to the end of the input, if it found another "split by" it'll rewrite it without it. I'm not sure if yours would be able to do something like test1-split-test2 by -split-, but I could be wrong. It's also very strange because I didn't want to use any variables, and there could be another custom block using the list that would effect where all the data is placed

**Comment by softedco** [comment]
> My point was if you were to split say test1-split-test2 by -split-, it would get stuck searching for a -split- as a single character. My code was altered to check if letter one matches, then letter 2, then letter 3, etc, and it resets itself if the next letter doesn't match. Once you get all the letters matched, it changes what would essentially be the "item" by -1, and once it reaches 0 it starts to actually keep track of what it goes over. Once it gets to -1, due to it recording the next instance of a "split by" or gets to the end of the input, if it found another "split by" it'll rewrite it without it. I'm not sure if yours would be able to do something like test1-split-test2 by -split-, but I could be wrong. It's also very strange because I didn't want to use any variables, and there could be another custom block using the list that would effect where all the data is placed

I get what you're saying and I've tried that before, I think there is an effective enough way to approach this

about other blocks affecting each other, scratch should be thread safe enough to prevent blocks from interrupting each other so that is not an issue, a single retval variable for returning blocks should work also because of that
